src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdAuthenticatorFactory.java

```diff
@@ -3,40 +3,36 @@
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.stereotype.Service;
 import uk.gov.hmcts.reform.authorisation.generators.AuthTokenGenerator;
-import uk.gov.hmcts.reform.bulkscan.orchestrator.services.idam.Credential;
 import uk.gov.hmcts.reform.bulkscan.orchestrator.services.idam.JurisdictionToUserMapping;
-import uk.gov.hmcts.reform.idam.client.IdamClient;
-import uk.gov.hmcts.reform.idam.client.models.UserDetails;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.idam.cache.CachedIdamCredential;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.idam.cache.IdamCachedClient;
 
 @Service
 @EnableConfigurationProperties(JurisdictionToUserMapping.class)
 public class CcdAuthenticatorFactory {
 
     private final AuthTokenGenerator s2sTokenGenerator;
-    private final IdamClient idamClient;
-    private final JurisdictionToUserMapping users;
+    private final IdamCachedClient idamClient;
 
     public CcdAuthenticatorFactory(
         AuthTokenGenerator s2sTokenGenerator,
-        IdamClient idamClient,
-        JurisdictionToUserMapping users
+        IdamCachedClient idamClient
     ) {
         this.s2sTokenGenerator = s2sTokenGenerator;
         this.idamClient = idamClient;
-        this.users = users;
     }
 
     public CcdAuthenticator createForJurisdiction(String jurisdiction) {
-        Credential user = users.getUser(jurisdiction);
-        String userToken = idamClient.authenticateUser(user.getUsername(), user.getPassword());
-        UserDetails userDetails = idamClient.getUserDetails(userToken);
+        CachedIdamCredential idamCredentials = idamClient.getIdamCredentials(jurisdiction);
 
-        //TODO: RPE-738 the userToken needs a to be cached and timed-out.
-        // this can be decorated here like the s2sTokenGenerator
         return new CcdAuthenticator(
             s2sTokenGenerator::generate,
-            userDetails,
-            () -> userToken
+            idamCredentials.userDetails,
+            () -> idamCredentials.accessToken
         );
     }
+
+    public void removeFromCache(String jurisdiction) {
+        idamClient.removeAccessTokenFromCache(jurisdiction);
```

Comment:
Woud it be best if idamClient's token management itself was case-insensitive?

Reply:
may be I did not understand but idam does not know anything about juristidiction

Reply:
Sorry, my bad. IdamClient's token management is case-insensitive.